### ISSUE TITLE
best way to do a one-time provide?

### ISSUE THREAD

**Issue by rpodgorny** [ipfs/go-ipfs#7507]
i'd like to share a (ptentially huge) set of files.

1) this would be a one-time operation - i'd quit the providing when the other side is done downloading (think torrents).
2) the data amount is huge - there should be no need to import it all to local node's datastore (there even may be no local node at all).
3) i don't want to clobber with the local node at all (so no filestore?).
4) it's fine when the start takes a long time (for the hashing and dht stuff).
5) it's fine when it takes some (a lot?) of memory for the internal data structures.

in other words:

1) server: ipprovide /some/path
2) server: starting node abc
3) server: ...crunching...
4) server: cid is xyz
5) client: ipget xyz
6) client: coffee...
7) client: ...finished
8) server: ctrl-c

what would be the best way to achieve this? i've tried to google it extensively but i've found no suitable solution... :-(

**Comment by aschmahmann** [comment]
@rpodgorny there is not currently a way to specify that a specific DAG should not be reprovided. However, you can change the reprovider strategy for the entire node or disable reproviding entirely as described in the config docs https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#reprovider.

Note: If you're trying to get data from a known server to some clients then you can skip the initial provide as well by passing the `--offline` flag to `ipfs add` and then have peers manually connect (i.e. `ipfs swarm connect /p2p/QmYourServer`) before asking for data. This will trigger Bitswap to ask for the data from your server even though it was never advertised in the DHT.

**Comment by rpodgorny** [comment]
> @rpodgorny there is not currently a way to specify that a specific DAG should not be reprovided. However, you can change the reprovider strategy for the entire node or disable reproviding entirely as described in the config docs https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#reprovider.
> 
> Note: If you're trying to get data from a known server to some clients then you can skip the initial provide as well by passing the `--offline` flag to `ipfs add` and then have peers manually connect (i.e. `ipfs swarm connect /p2p/QmYourServer`) before asking for data. This will trigger Bitswap to ask for the data from your server even though it was never advertised in the DHT.

i'm afraid i don't understand how this relates to what i'm trying to achieve. i've probably not expressed myself clearly. :-( let me try with other words:

what would be the opposite program to "ipget" (wget-like util for a one-time download from ipfs)? a tool that would just one-time "publish" some content (directory, recursively) to the ipfs network? it would not share any state/config/data with the currently running ipfs daemon...

**Comment by aschmahmann** [comment]
@rpodgorny I think your question belongs on discuss.ipfs.io since it's a general support question and not a bug/feature request from go-ipfs. As nobody really searches through closed github issues to learn how to do things posting to a forum will allow people with similar questions to you to find the results.

I'll take a look at the forums later today to respond to your post.
